### DISCUSSION TITLE
Does Tart support stateful VM's? (saving snapshots)

### DISCUSSION THREAD

**Discussion by GuyDev1** [cirruslabs/tart#485] [Q&A] [answered]
Hey,
Is there a way to save the state of the VM? so we don't start as a clean slate every time?

Thanks!

**Comment by fkorotkov** [comment, marked as answer]
Tart is not supporting live snapshotting in the classical sense but you can always `stop` a VM in a desired state, `clone` it to preserve a "snapshot" of the current state and than you can resume `run`ning it.

FYI `clone` command is just coping of the disk image which is super fast and disk efficient due to magic of APFS which will only physically store difference between snapshotted disk and the VM's disk.

Please subscribe to updates of #147 if Tart will ever support live snapshotting of running VMs.

**Reply by GuyDev1** [reply]
@fkorotkov Nice, so essentially I can start the VM, prepare it, and then after stopping it (but not closing it completely), clone it by running `tart clone old_vm_name new_vm_name` and when running the new cloned VM, it has the state of the old one. Cool!

Thanks!

**Reply by fkorotkov** [reply]
You need to shutdown the VM completely for the clone but pretty much yes. Please also be aware of a [plugin for Packer](https://github.com/cirruslabs/packer-plugin-tart) that simplifies repeatable building of VMs.

**Reply by GuyDev1** [reply]
Yeah just did that and it worked :) Thanks
